src/Diagnostics.RuntimeHost/Services/SearchService/SearchService.cs

```diff
@@ -30,13 +31,16 @@ public class SearchService : ISearchService
         private string TriggerTrainingUrl;
         private string TriggerModelRefreshUrl;
         private HttpClient _httpClient;
+        SearchServiceProviderConfiguration configuration;
 
-        public SearchService()
+        public SearchService(IServiceProvider services)
```

Comment:
Inject `IDataSourcesConfigurationService` instead of using service locator pattern